sumpy/e2e.py

```diff
@@ -600,17 +583,22 @@ def get_kernel(self, result_dtype):
                 name=self.name,
                 assumptions="ntranslation_classes>=1",
                 default_offset=lp.auto,
-                fixed_parameters=dict(dim=self.dim),
+                fixed_parameters=dict(
+                    dim=self.dim,
+                    m2l_translation_classes_dependent_ndata=(
+                        m2l_translation_classes_dependent_ndata)),
                 lang_version=MOST_RECENT_LANGUAGE_VERSION
                 )
 
         for knl in [self.src_expansion.kernel, self.tgt_expansion.kernel]:
             loopy_knl = knl.prepare_loopy_kernel(loopy_knl)
 
-        loopy_knl = lp.tag_inames(loopy_knl, "idim*:unr")
+        loopy_knl = lp.merge([loopy_knl, child_knl])
+        loopy_knl = lp.inline_callable_kernel(loopy_knl, "m2l_data_inner")
         loopy_knl = lp.set_options(loopy_knl,
                 enforce_variable_access_ordered="no_check")
 
+        loopy_knl = lp.tag_inames(loopy_knl, "idim*:unr")
```

Comment:
Justify (maybe via a comment) that this isn't just an optimization. (If it is, it should be in `get_optimized_kernel`.

Reply:
Moved to `get_optimized_kernel` in 7d9f5359626ea40c7a0ede7afb8c4a55224b1fb9